### ISSUE TITLE
Infinite recursion in AlternatingDegree of a non-natural alternating group constructed with AlternatingGroup

### ISSUE THREAD

**Issue by sTertooy** [gap-system/gap#4826]
### Observed behaviour

```
gap> A := AlternatingGroup( IsPcGroup, 4 );;
gap> AlternatingDegree( A );
Error, recursion depth trap (5000) in
  AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
AlternatingDegree( G ) at /usr/lib/gap/lib/grpnames.gi:1205 called from
...  at *stdin*:16
```
The problem seems to lie here:
```
gap> HasIsAlternatingGroup( A ) and IsAlternatingGroup( A );
true
gap> IsNaturalAlternatingGroup( A );
false
gap> Print( ApplicableMethod( AlternatingDegree, [A] ) );
function ( G )
    if not IsFinite( G ) then
        TryNextMethod();
    fi;
    if IsNaturalAlternatingGroup( G ) then
        return DegreeAction( G );
    fi;
    if IsAlternatingGroup( G ) then
        return AlternatingDegree( G );
    else
        return fail;
    fi;
    return;
end
```
So, if a group has `IsAlternatingGroup` set to `true` but does not have the  `AlternatingDegree` attribute set, we get an infinite recursion.
However, if the group doesn't have `IsAlternatingGroup` set yet, calling it will also set the `AlternatingDegree` attribute, avoiding this bug:
```
gap> B := AlternatingGroup( IsPermGroup, 4 );;
gap> C := Image( IsomorphismPcGroup( B ) );;
gap> HasIsAlternatingGroup( C ) or HasAlternatingDegree( C );
false
gap> IsAlternatingGroup( C ) and HasAlternatingDegree( C );
true
gap> AlternatingDegree( C );
4
```

### Expected behaviour
Ideally, we would get the alternating degree.
```
gap> A := AlternatingGroup( IsPcGroup, 4 );;
gap> AlternatingDegree( A );
4
```

### Copy and paste GAP banner (to tell us about your setup)
```
 ┌───────┐   GAP 4.12dev-1220-g516e42e built on 2022-03-09 15:58:57+0100
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-pc-linux-gnu-default64-kv8
 Configuration:  gmp 6.2.0, GASMAN, readline
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.5, PrimGrp 3.4.1, SmallGrp 1.4.2, TransGrp 3.3
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
```
The same error is already present in GAP 4.9.3.

**Comment by fingolfin** [comment]
Thanks for the report! The same issues probably affects `SymmetricDegree` as well.

**Comment by fingolfin** [comment]
Fix in PR #4830
